pkg/vex/vex.go

```diff
@@ -145,14 +145,9 @@ func (vexDoc *VEX) EffectiveStatement(product, vulnID string) (s *Statement) {
 	SortStatements(statements, t)
 
 	for i := len(statements) - 1; i >= 0; i-- {
-		if statements[i].Vulnerability.ID != vulnID {
+		if !statements[i].Matches(vulnID, product, nil) {
 			continue
```

Comment:
I think this needs to return the statement when it finds a match :)

Something like:

```suggestion
		if statements[i].Matches(vulnID, product, nil) {
			return &statements[i]
```